src/v/cluster/metrics_reporter.h

```diff
@@ -69,6 +69,7 @@ class metrics_reporter {
         uint32_t partition_count;
 
         std::vector<node_metrics> nodes;
+        bool has_kafka_gssapi;
```

Comment:
since there is a creation like this

```
    metrics_snapshot snapshot;
```

will we end up having the sanitizer issue where it complains bool doesn't contain 0 or 1?

Reply:
Only if it's read before being written. It's unconditionally written before being read.

Reply:
ah good to know. i thought it was catching the issue every time there was a construction.